### ISSUE TITLE
Check and automatically (or prompt) to update the path of missing images/shaders/etc. if the files are found in a subdirectory or configured media path

### ISSUE THREAD

**Issue by kevinsaucier** [smeighan/xLights#4173]
When moving sequences from my 2022 folder to my 2023 folder, I open the sequence and it prompts that the path (pointing to the 2022 folder) for images no longer exists and I get the dreaded red box.  In reality, the files exist in the same folder (or a subfolder) as the sequence but xLights doesn't seem to look for them.  

It would be ideal if xLights stored a 'relative' path for files that exist within the same/sub directory as the sequence (i.e. .\Images\image.jpg) so the entire sequence folder could be moved at will.  It would be further ideal if xLights would search the sequence/sub directory for missing files (images/shaders/etc.) and the configured media directories and either automatically update effects with incorrect entries or prompt to update them.

I thought this was how it already worked (could have sworn I'd seen that somewhere) but I've run across the same issue multiple times while migrating over last years' sequences to this year.  I know I still have numerous sequences pointing at wrong folders and I expect I'll find that as time goes on and I rename/remove those old folders.

I can share a sequence with this issue but they're paid sequences so I don't want to just post them.  Please let me know if you need a sample sequence.

Thanks!

**Comment by kevinsaucier** [comment]
Yup, that's what didn't work.  

<root>\2022\<SequenceName>\  contains the xsq and a subfolder called Images with the image files.

Moved <SequenceName> to <root>\2023 (so it's <root>\2023\<SequenceName> and got prompted that the files didn't exist.  It's happened on multiple sequence folders.

**Comment by AzGilrock** [comment]
Yeah quit putting them into a Videos subfolder.  Put the stuff the sequence needs in the same folder as the sequence.

**Comment by computergeek1507** [comment]
If you put them one folder deep of your show folder i.e showfolder/videos it will pick them up. I personally don't understand the making a new show folder ever year, I use the same folder every year. I never plan to put up my 2018 layout again so why keep it.

**Comment by AzGilrock** [comment]
> > Yeah quit putting them into a Videos subfolder. Put the stuff the sequence needs in the same folder as the sequence.
> 
> I 100% cannot do that. It will literally kill me. 😁

Quit being over dramatic.  Select them all and move them up one level.  Then edit the sequence XML and batch fix the filepath which it appears your going to need to do anyways.

**Comment by kevinsaucier** [comment]
I'm really not trying to be argumentative, I promise, but this seems like a usability bug that could be easily resolved by adding a "/recurse" (or the equivalent) to the search command.  I mean, really, even if I move the files up a level, so they're even with the XSQ, xLights still stores the full path in the XML.  What's it hurt to store the full path to a subfolder (though I still think it should store the path relative to the XSQ).  Oh well, to dream......

I'd be all for adding a PR but this is not my language and, quite honestly, Keith's description of how to set up the development environment scared me off.  😁

**Comment by AzGilrock** [comment]
We used to require all files for every sequence to exist in the show folder.   It looks to me like you added Christmas to the folder path.   Not sure but that extra level could have thrown it off.

**Comment by kevinsaucier** [comment]
Side note, I just noticed that xLights seems to create the same directory structure I'm using when you import from a ZIP.  I never really use the ZIP import, but I just did and it gave me this:

![image](https://github.com/smeighan/xLights/assets/25715814/af5f411f-a1c1-4c4c-88f7-d1708fe048c0)


I checked the original ZIP (from the vendor) and it only had an Images subfolder with both the PNG/JPG and MP4 files in it.  So, seems like xLights is creating a file structure that no longer works when the folder is moved.

**Comment by AzGilrock** [comment]
You know what I’ve done?  Package my own sequence then Import it into my new folder.  Fixed up all the paths.

**Comment by derwin12** [comment]
So much has changed now .. we now store relative paths in media. There are also options to change the paths in the media panel along with it now saving locations if it can find it on your media paths.
